R/sample.R

```diff
@@ -156,16 +213,62 @@ combine_chains <- function(res) {
                 "runner")))
   }
 
-  ret <- list(pars = pars,
-              density = density,
-              details = details,
-              chain = chain)
-  class(ret) <- "mcstate_samples"
-  ret
+  samples <- list(pars = pars,
+                  density = density,
+                  details = details,
+                  chain = chain)
+  class(samples) <- "mcstate_samples"
+  samples
+}
+
+
+## This is absolutely terrible, but it will get there.
+append_chains <- function(prev, curr) {
+  n_chains <- length(prev$restart$rng_state)
+  i <- split(seq_along(prev$chain), factor(prev$chain, seq_len(n_chains)))
+  j <- lapply(
+    split(seq_along(curr$chain), factor(curr$chain, seq_len(n_chains))),
+    function(x) x[-1] + length(prev$chain))
+  k <- unlist(matrix(c(i, j), 2, byrow = TRUE))
```

Comment:
completely optional but i think it's less gross, also why are we skipping that first step of the restarted chain? is it because that'll be the initial so it will be repeated?
```suggestion
  i <- split(seq_along(prev$chain), prev$chain)
  j <- split(seq_along(curr$chain) + length(prev$chain), curr$chain)
  j <- lapply(j, function(x) x[-1])
  k <- unlist(rbind(i, j))
```

Reply:
> why are we skipping that first step of the restarted chain? is it because that'll be the initial so it will be repeated

Yes - exactly that. Ed and I are discussing if the first step is really part of the chain anyway; we might decide to discard it an retain initial conditions instead, in which case this will simplify slightly